File: platform/common-lib/src/main/java/com/researchex/common/error/BusinessException.java
package com.researchex.common.error;

import lombok.Getter;

/** 서비스 도메인에서 발생하는 예외의 공통 부모 클래스. */
@Getter
public class BusinessException extends RuntimeException {

  // 예외에 매핑된 도메인 에러 코드를 노출한다.
  private final ErrorCode errorCode;

  public BusinessException(ErrorCode errorCode) {
    super(errorCode.getMessage());
    this.errorCode = errorCode;
  }

  public BusinessException(ErrorCode errorCode, String message) {
    super(message);
    this.errorCode = errorCode;
  }

  public BusinessException(ErrorCode errorCode, String message, Throwable cause) {
    super(message, cause);
    this.errorCode = errorCode;
  }
}

File: platform/common-lib/src/main/kotlin/com/researchex/common/error/BusinessException.kt
package com.researchex.common.error

/** 서비스 도메인에서 발생하는 예외의 공통 부모 클래스다. */
open class BusinessException(
    val errorCode: ErrorCode,
    message: String = errorCode.message,
    cause: Throwable? = null
) : RuntimeException(message, cause)
